tests/Unit/Structs/Soa/v1/AddSignerRequestTest.php

```diff
@@ -0,0 +1,87 @@
+<?php declare(strict_types=1);
+
+namespace Coverzen\Components\YousignClient\Tests\Unit\Structs\Soa\v1;
+
+use Coverzen\Components\YousignClient\Enums\v1\SignatureAuthenticationMode;
+use Coverzen\Components\YousignClient\Enums\v1\SignatureLevel;
+use Coverzen\Components\YousignClient\Exceptions\Structs\v1\StructSaveException;
+use Coverzen\Components\YousignClient\Structs\Soa\v1\AddSignerRequest;
+
+/**
+ * Class AddSignerRequestTest.
+ *
+ * @coversDefaultClass \Coverzen\Components\YousignClient\Structs\Soa\v1\AddSignerRequest
+ */
+final class AddSignerRequestTest extends TestCase
+{
+    /**
+     * @test
+     *
+     * @return void
+     */
+    public function it_instantiates(): void
+    {
+        $addSignerRequest = new AddSignerRequest();
+
+        $this->assertInstanceOf(AddSignerRequest::class, $addSignerRequest);
+    }
+
+    /**
+     * @test
+     * @covers \Coverzen\Components\YousignClient\Structs\Soa\v1\AddSignerRequest::factory
+     *
+     * @return void
+     */
+    public function it_is_made_by_factory(): void
+    {
+        /** @var AddSignerRequest $addSignerRequest */
+        $addSignerRequest = AddSignerRequest::factory()
+                                            ->make();
+
+        $this->assertInstanceOf(AddSignerRequest::class, $addSignerRequest);
+    }
+
+    /**
+     * @test
+     *
+     * @return void
+     */
+    public function it_has_expected_properties(): void
```

Comment:
Questa è piu una cosa generica per @giraz82 
In questo modo (che quello che abbiamo sempre usato) io in realta sto testando che la factory mi costruisca il modello nel modo che idealmente io ritendo giusto e che ho dichiarato nelle properties della classe. 

Ma in realta non ho nessun type hint reale che mi dice che una prop `@prop strint|null $pippo ` accetti effettivametne solo stringhe, perche se io faccio $classe->pippo = [] non ho errori.

Reply:
Sì, sono cose che normalmente scrivo quando parto da 0 per lavorare in TDD.
Però questo, unitamente agli static analyzers che controllano anche i tipi dichiarati nel phpdoc, ti da un a discreta sicurezza.
Se tu fai `$classe->pippo = []` non hai errori nei test, ma phpstan e psalm non te la danno buona.